src/libutil/url.cc

```diff
@@ -79,10 +79,15 @@ std::map<std::string, std::string> decodeQuery(const std::string & query)
 
     for (auto s : tokenizeString<Strings>(query, "&")) {
         auto e = s.find('=');
-        if (e != std::string::npos)
-            result.emplace(
-                s.substr(0, e),
-                percentDecode(std::string_view(s).substr(e + 1)));
+
+        if (e == std::string::npos) {
+            warn("invalid URI query '%s', did you forget an equals sign `=`?", s);
```

Comment:
```suggestion
            warn("dubious URI query '%s' is missing equal sign '%s'", s, "=");
```

- if truly invalid, we'd have to throw an error
- maybe the URL wasn't input by this user
- `%s` are highlighted, representing user input and code